File: backend/src/main/java/com/neu/hotel/entity/Review.java
package com.neu.hotel.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 评价实体类
 *
 * 【数据库表】biz_review - 评价表
 *
 * 【主要功能】
 * - 记录客人入住后的评价信息
 * - 支持评价回复功能
 * - 为酒店服务改进提供依据
 *
 * 【字段说明】
 * - id: 评价唯一标识（自增主键）
 * - checkInId: 关联的入住记录ID（对应 biz_check_in 表）
 * - userId: 评价人ID（对应 sys_user 表）
 * - roomId: 入住房间ID（对应 biz_dorm_room 表）
 * - rating: 评分（1-5星）
 * - content: 评价内容
 * - reply: 管理员回复内容
 * - replyTime: 回复时间
 * - replyUserId: 回复人ID
 * - roomNo/checkInDate/checkOutDate: 非数据库字段，用于展示
 *
 * 【关联关系】
 * - 通过 checkInId 关联 CheckIn 实体
 * - 通过 roomId 关联 DormRoom 实体
 * - 通过 ReviewService 处理评价业务逻辑
 * - 通过 ReviewController 提供评价管理API
 *
 * 【对应前端】
 * - 路由：/admin/review（后台管理）
 * - 路由：/portal/review（前台展示）
 * - API：@/api/review.js
 * - Vue组件：@/views/visitor/review/index.vue
 */
@Data
@TableName("biz_review")
public class Review implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;              // 评价唯一标识

    private Long checkInId;       // 关联入住记录ID

    private Long userId;         // 评价人ID

    private Long roomId;         // 房间ID

    private Integer rating;       // 评分

    private String content;       // 评价内容

    private String reply;         // 回复内容

    private LocalDateTime replyTime;  // 回复时间

    private Long replyUserId;     // 回复人ID

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;  // 创建时间

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;  // 更新时间

    // ========== 非数据库字段（用于关联查询和展示） ==========

    @TableField(exist = false)
    private String roomNo;        // 房间号

    @TableField(exist = false)
    private String checkInDate;   // 入住日期

    @TableField(exist = false)
    private String checkOutDate;  // 退房日期

    @TableLogic
    private Integer deleted;       // 逻辑删除标记
}

File: frontend/src/api/review.js
/**
 * 评价管理相关API接口
 *
 * 【模块说明】
 * - 处理客人入住后的评价
 * - 管理评价回复
 *
 * 【接口列表】
 * 1. submitReview: 提交评价
 * 2. getMyReview: 获取我的评价
 * 3. getCheckInReview: 获取订单评价
 * 4. getRecentReviews: 获取最新评价
 * 5. getRoomReviews: 获取房间评价
 * 6. replyReview: 回复评价
 *
 * 【对应后端】
 * - Controller: ReviewController
 * - 路径前缀: /review
 *
 * 【对应前端组件】
 * - 评价管理: @/views/visitor/review/index.vue
 */

import request from '@/utils/request'

/**
 * 提交评价
 * @param {Object} data - 评价信息 { checkInId, rating, content }
 * @returns {boolean} 是否成功
 */
export function submitReview(data) {
  return request({
    url: '/review',
    method: 'post',
    data
  })
}

/**
 * 获取当前用户对某订单的评价
 * @param {number} checkInId - 订单ID
 * @returns {Object} 评价信息
 */
export function getMyReview(checkInId) {
  return request({
    url: '/review/my',
    method: 'get',
    params: { checkInId }
  })
}

/**
 * 获取某订单的评价
 * @param {number} checkInId - 订单ID
 * @returns {Object} 评价信息
 */
export function getCheckInReview(checkInId) {
  return request({
    url: `/review/checkin/${checkInId}`,
    method: 'get'
  })
}

/**
 * 管理员回复评价
 * @param {number} id - 评价ID
 * @param {string} reply - 回复内容
 * @returns {boolean} 是否成功
 */
export function replyReview(id, reply) {
  return request({
    url: `/review/reply/${id}`,
    method: 'put',
    params: { reply }
  })
}

/**
 * 获取最新评价列表（用于首页展示）
 * @param {number} limit - 返回数量（默认6）
 * @returns {Array} 评价列表
 */
export function getRecentReviews(limit = 6) {
  return request({
    url: '/review/recent',
    method: 'get',
    params: { limit }
  })
}

/**
 * 获取某房间的所有评价
 * @param {number} roomId - 房间ID
 * @returns {Array} 评价列表
 */
export function getRoomReviews(roomId) {
  return request({
    url: `/review/room/${roomId}`,
    method: 'get'
  })
}
